packages/grafana-ui/src/components/Table/HeaderRow.tsx

```diff
@@ -43,14 +45,22 @@ export const HeaderRow = (props: HeaderRowProps) => {
 
 function renderHeaderCell(column: any, tableStyles: TableStyles, showTypeIcons?: boolean) {
   const headerProps = column.getHeaderProps();
-  const field = column.field ?? null;
+  const field: Field = column.field ?? null;
+  const tableFieldOptions: TableFieldOptions | undefined = field?.config.custom;
 
   if (column.canResize) {
     headerProps.style.userSelect = column.isResizing ? 'none' : 'auto'; // disables selecting text while resizing
   }
 
   headerProps.style.position = 'absolute';
   headerProps.style.justifyContent = column.justifyContent;
+  headerProps.style.left = column.left;
+
+  let headerContent = column.render('Header');
+
+  if (tableFieldOptions?.headerComponent) {
```

Comment:
Rendering the header content within the button doesn't play nice when sort is enabled, I think we'll want to update the defaultContent prop to contain everything within the table, not just the title within the button, for example:

```
  let headerContent = column.render('Header');

  let sortHeaderContent = column.canSort && (
      <>
        <button {...column.getSortByToggleProps()} className={tableStyles.headerCellLabel}>
          {showTypeIcons && (
            <Icon name={getFieldTypeIcon(field)} title={field?.type} size="sm" className={tableStyles.typeIcon} />
          )}
          <div>{headerContent}</div>
          {column.isSorted &&
            (column.isSortedDesc ? (
              <Icon size="lg" name="arrow-down" className={tableStyles.sortIcon} />
            ) : (
              <Icon name="arrow-up" size="lg" className={tableStyles.sortIcon} />
            ))}
        </button>
        {column.canFilter && <Filter column={column} tableStyles={tableStyles} field={field} />}
      </>
    )
  if(sortHeaderContent && tableFieldOptions?.headerComponent){
    sortHeaderContent = <tableFieldOptions.headerComponent field={field} defaultContent={sortHeaderContent} />;
  }else if (tableFieldOptions?.headerComponent) {
    headerContent = <tableFieldOptions.headerComponent field={field} defaultContent={headerContent} />;
  }

  return (
    <div className={tableStyles.headerCell} {...headerProps} role="columnheader">
      {column.canSort && sortHeaderContent}
      {!column.canSort && headerContent}
      {!column.canSort && column.canFilter && <Filter column={column} tableStyles={tableStyles} field={field} />}
      {column.canResize && <div {...column.getResizerProps()} className={tableStyles.resizeHandle} />}
    </div>
  );
```